pkg/block/metadata/hash.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package metadata
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// HashKind indicates what type of hash it is.
+type HashKind string
```

Comment:
It's not kind but maybe `function`?

```suggestion
type HashFunc string
```
or maybe type?